pandas/_libs/tslibs/tzconversion.pyx

```diff
@@ -375,6 +335,70 @@ cdef inline str _render_tstamp(int64_t val):
     return str(Timestamp(val))
 
 
+cdef ndarray[int64_t] _get_dst_hours(
```

Comment:
Would be good to reference https://github.com/pandas-dev/pandas/issues/45905 which is a request to make this work if the times are monotonically decreasing

Reply:
will write myself a note to that effect, not sure that is actionable for this PR